### PR TITLE
Add 3.0-to-3.1 migration doc

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/15678

[Internal review page](https://review.docs.microsoft.com/aspnet/core/migration/30-to-31?view=aspnetcore-1.0&branch=pr-en-us-15875&tabs=visual-studio)

### CODE REVIEW

File: aspnetcore/migration/30-to-31.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+[!INCLUDE[](~/includes/net-core-prereqs-mac-3.1.md)]
+
+---
+
+## Update .NET Core SDK version in global.json
+
+If you rely upon a [global.json](/dotnet/core/tools/global-json) file to target a specific .NET Core SDK version, update its `version` property to the 3.1 version installed on your machine. For example:
```
Comment (Rick-Anderson) [inline aspnetcore/migration/30-to-31.md]: If you rely upon a [global.json](/dotnet/core/tools/global-json) file to target a specific .NET Core SDK version, update the `version` property to the installed SDK version. For example:

File: aspnetcore/migration/30-to-31.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  "sdk": {
+-    "version": "3.0.101"
++    "version": "3.1.100-preview3-014645"
+  }
+}
+```
+
+## Update the target framework
+
+In your project file, update the [Target Framework Moniker (TFM)](/dotnet/standard/frameworks) to `netcoreapp3.1`:
```
Comment (Rick-Anderson) [inline aspnetcore/migration/30-to-31.md]: In the project file, update the [Target Framework Moniker (TFM)](/dotnet/standard/frameworks) to `netcoreapp3.1`:

File: aspnetcore/migration/30-to-31.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+-    <TargetFramework>netcoreapp3.0</TargetFramework>
++    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+</Project>
+```
+
+## Update package references
+
+In your project file, update each `Microsoft.AspNetCore.*` package reference's `Version` attribute to 3.1.0 or later. For example:
```
Comment (Rick-Anderson) [inline aspnetcore/migration/30-to-31.md]: In the project file, update each `Microsoft.AspNetCore.*` package reference's `Version` attribute to 3.1.0 or later. For example:

File: aspnetcore/migration/30-to-31.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0" />
+-    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation" Version="3.0.0" Condition="'$(Configuration)' == 'Debug'" />
++    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.0-preview3.19555.2" />
++    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation" Version="3.1.0-preview3.19555.2" Condition="'$(Configuration)' == 'Debug'" />
+</ItemGroup>
+```
+
+## Update Docker images
+
+If your app is using Docker, use a base image that includes ASP.NET Core 3.1. For example:
```
Comment (Rick-Anderson) [inline aspnetcore/migration/30-to-31.md]: For apps using Docker, use a base image that includes ASP.NET Core 3.1. For example:

File: aspnetcore/migration/30-to-31.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/migration/30-to-31.md]: I think the presence of the 3.1 is important in the sentence. The project won't compile with an earlier SDK.

Comment (Rick-Anderson) [reply aspnetcore/migration/30-to-31.md]: Sure, add it in.

File: aspnetcore/migration/30-to-31.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+## Update .NET Core SDK version in global.json
+
+If you rely upon a [global.json](/dotnet/core/tools/global-json) file to target a specific .NET Core SDK version, update the `version` property to the 3.1 SDK version that's installed. For example:
+
+```diff
+{
+  "sdk": {
+-    "version": "3.0.101"
++    "version": "3.1.100-preview3-014645"
```
Comment (danroth27) [inline aspnetcore/migration/30-to-31.md]:
```suggestion
+    "version": "3.1.100"
```

Comment (scottaddie) [reply aspnetcore/migration/30-to-31.md]: I plan to update all version numbers in the doc to RTM with the following issue: https://github.com/aspnet/AspNetCore.Docs/issues/15884. This PR uses the Preview 3 version numbers.

**Pull request merged**